doc/user_guide/psyir.rst

```diff
@@ -134,3 +185,34 @@ in the PSyIR. At the moment the available symbols are:
 
 - .. autoclass:: psyclone.psyir.symbols.DataSymbol
 
+Creating or adding to PSyIR
+---------------------------
+
+PSyIR nodes are connected together via parent and child methods
+provided by the ``Node`` baseclass.
+
+These nodes can be created in isolation and then connected
+together. For example::
+
+    literal = Literal("0.0")
+    reference = Reference("a")
+    assignment = Assignment()
+    literal.parent = assignment
+    reference.parent = assignment
+    assignment.children = [reference, assignment]
+
+However, as connections get more complicated, creating the correct
+connections can become difficult to manage and error prone. Further,
+in some cases children must be collected together within a
+``Schedule`` (e.g. for ``IfBlock`` and for ``Loop``).
+
+To simplify this complexity, each of the Kernel-layer nodes which
+contain other nodes have a ``create`` method which helps construct the
+PSyIR. Using this method, the above example then becomes::
+  
+    literal = Literal("0.0")
+    reference = Reference("a")
+    assignment = Assignment.create(reference, literal)
+
+A more complete example of using this approach can be found in the
+PSyclone ``examples/psyir`` directory.
```

Comment:
Do we need to mention that the default constructor can be used for the top-down approach (kinda)?

Reply:
As agreed, we're going to keep quiet about this in the user guide as the top down approach is currently rather messy so we don't want to promote it :-)